modules/navigation/3d/nav_mesh_generator_3d.cpp

```diff
@@ -700,7 +700,7 @@ void NavMeshGenerator3D::generator_bake_from_source_geometry_data(Ref<Navigation
 	cfg.detailSampleDist = MAX(p_navigation_mesh->get_cell_size() * p_navigation_mesh->get_detail_sample_distance(), 0.1f);
 	cfg.detailSampleMaxError = p_navigation_mesh->get_cell_height() * p_navigation_mesh->get_detail_sample_max_error();
 
-	if (p_navigation_mesh->get_border_size() > 0.0 && !Math::is_equal_approx(p_navigation_mesh->get_cell_size(), p_navigation_mesh->get_border_size())) {
+	if (p_navigation_mesh->get_border_size() > 0.0 && Math::fmod((float)p_navigation_mesh->get_border_size(), (float)p_navigation_mesh->get_cell_size()) != 0.0) {
```

Comment:
The casts are redundant because both of these already return `float`.

Part of me wonders if this should be `real_t` to support navigation meshes that are bigger than a few kilometers, but this should be left to another PR. For this PR, I think the casts would only be harmful if we ever do change the return types to `real_t`, so I would just remove the casts.